2 3 daten speichern

Miishale opened the pull request:
User nach MVP
Speichern der Datenbanken umgesetzt
statische Mockdaten entfernt

Miishale committed:
erster Entwurf

Miishale committed:
Merge remote-tracking branch 'origin/develop' into 2-3-Daten_speichern

# Conflicts:
#	src/main/java/ch/bfh/bti7081/model/manager/SeminarManager.java

Miishale committed:
MVP umgesetzt

Miishale committed:
Merge remote-tracking branch 'origin/develop' into 2-3-Daten_speichern

Miishale committed:
Merge remote-tracking branch 'origin/Task09' into 2-3-Daten_speichern

Miishale committed:
MVP umgesetzt

Miishale committed:
Merge branch 'develop' into 2-3-Daten_speichern

Miishale committed:
Speichern umgesetzt

Azarny commented on src/main/java/ch/bfh/bti7081/model/manager/FaqManager.java:
```diff
@@ -16,8 +21,8 @@
-    public FaqEntry createFaqEntry(FaqEntry entry) {
-        throw new IllegalArgumentException("Not implemented yet.");
+    public void createFaqEntry(FaqEntry entry) {
+        faqEntryRepository.save(entry);
```
evt vorher noch eine Validierung?

Azarny commented on src/main/java/ch/bfh/bti7081/model/manager/ForumManager.java:
```diff
@@ -18,8 +23,8 @@ public ForumEntryDTO getForumEntry(Integer id) {
-    public ForumEntryDTO createForumEntry(ForumEntryDTO forumEntryDTO) {
-        throw new IllegalArgumentException("Not implemented yet.");
+    public void createForumEntry(ForumEntry forumEntry) {
+        forumEntryRepository.save(forumEntry);
```
wie oben, Validierung fehlt (ausser wir lassen das im Moment sein, weil es Forum Zeugs ist)

Miishale replied:
Wird im Presenter und DTO Konvertierung gemacht, ist aber eigentlich nicht relevant, dachte einfach dann hat man das schon

Azarny commented on src/main/java/ch/bfh/bti7081/model/manager/SeminarManager.java:
```diff
@@ -76,8 +81,8 @@
-    public Seminar createSeminar(Seminar seminar) {
-        throw new IllegalArgumentException("Not implemented yet.");
+    public void createSeminar(Seminar seminar) {
+        seminarRepository.save(seminar);
```
Validierung? Oder ist die im Presenter?

Miishale replied:
wenn es ein Objekt bis zu einer SeminarEntity geschafft hat ist es fertig validiert

yaronwalter7 replied:
Mmh, aus meiner Sicht wäre es sinnvoll hier noch die Validierung aufzurufen... 
Wir haben doch nur für das die Methode validateSeminar? 
Diese wird ja scheinbar nur in den Test-Methoden verwendet... 
Grundsätzlich dürfen wir uns beim Speichern m.E. nicht rein auf die GUI-Validierung verlassen, sondern sollten sicherheitshalber im Backend diese auch durchführen...

Azarny replied:
da hat Yaron recht, alles vom GUI muss zwingend validiert werden bevor wir damit arbeiten. Aber ob die Validierung jetzt hier oder im Presenter stattfindet ist Geschmackssache... Würde vorschlagen, dass sie im Manager gemacht wird, weil dann die Validierung immer vor dem DB Aufruf gemacht wird

Miishale replied:
ah ja klar, i ha irgendwie im chopf gha dasi das schomau ufgrüeft ha

Azarny commented on src/main/java/ch/bfh/bti7081/model/dto/UserDTO.java:
```diff
@@ -0,0 +1,64 @@
+import java.util.List;
+
+public class UserDTO {
+
```
die ID fehlt, diese wird benötigt um unterscheiden zu können ob der Benutzer schon existiert oder nicht

Miishale replied:
Wird angepasst

Miishale replied:
has apasst, aber bi seminar und forumentry (die zwöi dtos wos scho git) hets o no ke ID... bruchts die dert oh?
zudem wird aktuell eh noni unterstützt neui user z spichere...

yaronwalter7 commented on src/main/java/ch/bfh/bti7081/model/manager/SeminarCategoryManager.java:
```diff
@@ -10,8 +12,11 @@
+        return seminarCategoryRepository.findAll();
     }
 
     public SeminarCategory getSeminarByName(String name) throws Exception {
```
Mmh, da du mit dem Repository bereits gearbeitet hast: Gibt es eine schnelle und einfache Möglichkeit diesen Code mit einem einzelnen Repository-Aufruf zu ersetzen?

Azarny replied:
Evt (ohne Gewähr): 
`seminarCategoryRepository.findByName(name);`

yaronwalter7 commented on src/main/java/ch/bfh/bti7081/model/manager/UserManager.java:
```diff
@@ -17,23 +23,12 @@ public static User getUserByUsername(String name) {
     }
 
+    public String encryptPassword(String password){
+        return "Not implemented yet";
```
Mmh, m.E. wäre es besser hier eine Exception zu werfen. (Z.B. NotImplementedException) 
So wird schlimmstenfalls Not Implement Yet als Passwort gespeichert und es fällt nicht auf...

Miishale replied:
Ds wird imne angere PR scho usprogrammiert requested

yaronwalter7 reviewed (approved):
Aus meiner Sicht, abgesehen von den gegebenen Kommentaren, ok.

Miishale committed:
added id in userDTO
readded clickable link (was merged away somehow)
added validation before saving seminaries

Miishale committed:
Zeilenlänge angepasst

Miishale committed:
Merge remote-tracking branch 'origin/develop' into 2-3-Daten_speichern

# Conflicts:
#	src/main/java/ch/bfh/bti7081/view/SeminarView.java

Miishale committed:
merged development
